hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotCreateRequest.java

```diff
@@ -210,5 +225,19 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     }
     return omClientResponse;
   }
-  
+
+  /**
+   * Removes the snapshot from the SnapshotChainManager.
+   */
+  private void removeSnapshotInfoFromSnapshotChainManager(
+      SnapshotChainManager snapshotChainManager,
+      SnapshotInfo info
+  ) {
+    try {
+      snapshotChainManager.deleteSnapshot(info);
+    } catch (IOException exception) {
+      LOG.warn("Failed to remove snapshot: {} from SnapshotChainManager.",
```

Comment:
If it fails to delete snapshot here, can be set and point latest global snapshot Id to previous successful snapshot by loading from db ?